resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -224,59 +260,7 @@ private[spark] class Client(
-                    " the latest state of the pod, another error was thrown. Check the logs for" +
-                    " the error that was thrown in looking up the driver pod.", e)
-              }
-              val topLevelMessage = s"The driver pod with name ${driverPod.getMetadata.getName}" +
-                s" in namespace ${driverPod.getMetadata.getNamespace} was not ready in" +
-                s" $driverLaunchTimeoutSecs seconds."
-              val podStatusPhase = if (driverPod.getStatus.getPhase != null) {
-                s"Latest phase from the pod is: ${driverPod.getStatus.getPhase}"
-              } else {
-                "The pod had no final phase."
-              }
-              val podStatusMessage = if (driverPod.getStatus.getMessage != null) {
-                s"Latest message from the pod is: ${driverPod.getStatus.getMessage}"
-              } else {
-                "The pod had no final message."
-              }
-              val failedDriverContainerStatusString = driverPod.getStatus
-                .getContainerStatuses
-                .asScala
-                .find(_.getName == DRIVER_LAUNCHER_CONTAINER_NAME)
-                .map(status => {
-                  val lastState = status.getState
-                  if (lastState.getRunning != null) {
-                    "Driver container last state: Running\n" +
-                    s"Driver container started at: ${lastState.getRunning.getStartedAt}"
-                  } else if (lastState.getWaiting != null) {
-                    "Driver container last state: Waiting\n" +
-                    s"Driver container wait reason: ${lastState.getWaiting.getReason}\n" +
-                    s"Driver container message: ${lastState.getWaiting.getMessage}\n"
-                  } else if (lastState.getTerminated != null) {
-                    "Driver container last state: Terminated\n" +
-                    s"Driver container started at: ${lastState.getTerminated.getStartedAt}\n" +
-                    s"Driver container finished at: ${lastState.getTerminated.getFinishedAt}\n" +
-                    s"Driver container exit reason: ${lastState.getTerminated.getReason}\n" +
-                    s"Driver container exit code: ${lastState.getTerminated.getExitCode}\n" +
-                    s"Driver container message: ${lastState.getTerminated.getMessage}"
-                  } else {
-                    "Driver container last state: Unknown"
-                  }
-                }).getOrElse("The driver container wasn't found in the pod; expected to find" +
-                  s" container with name $DRIVER_LAUNCHER_CONTAINER_NAME")
-              val finalErrorMessage = s"$topLevelMessage\n" +
-                s"$podStatusPhase\n" +
-                s"$podStatusMessage\n\n$failedDriverContainerStatusString"
+              val finalErrorMessage: String = getSubmitErrorMessage(kubernetesClient, e)
```

Comment:
maybe name this more like `constructDriverSubmitErrorMessage` -- `getX` makes me think there's no external calls being made, whereas this hits apiserver again looking for more detailed diagnostics